Added support to search for packages

crazyscientist opened the pull request:
This commit adds the required classes and functions to search for packages:
+ PackageCollection
+ ROPackage
+ find_package

marcus-h commented:
Hi,

first of all thanks for the PR!:) Below, a very minor nitpick:

On 2018-08-07 12:15:56 +0000, Andreas Pritschet wrote:
> class ROPackage(OscElement):
> +    """
> +    Represents a read only package.
> +
> +    This kind of package object is usually used in a collection.
> +    """

This is a bit inconsistent with the existing doc strings. Our current
format is

"""short summary

description

"""

(the same applies to the find_package function).
Can you adapt your patch accordingly?

PS: actually, I would complain about missing testcases, but since there
are no testcases for find_project as well I guess it is ok... (however,
patches with additional testcases are always welcome!;) )

PPS: out of curiosity, are you using/going to use the osc2 library?

crazyscientist commented:
Hi Marcus,

no problem, I have updated the docstrings.

Yes, the SMASH project at SUSE uses osc2 and I needed some information on packages. So I just copied and adjusted the function and class. In doing so I did not really look into test coverage or if the function should be made available on the CLI.

Assuming that SMASH will some day be migrated to Python3, I might find ample opportunity to contribute more code **and** tests ;)

marcus-h commented:
On 2018-08-08 12:17:53 +0000, Andreas Pritschet wrote:
> no problem, I have updated the docstrings.
> 
Almost;) at the end there is still a blank line missing
(sorry for being to super picky here...)

Also, can you squash the two commits into one?

> Yes, the SMASH project at SUSE uses osc2 and I needed some information on packages. So I just copied and adjusted the function and class. In doing so I did not really look into test coverage or if the function should be made available on the CLI.
> 
> Assuming that SMASH will some day be migrated to Python3, I might find ample opportunity to contribute more code **and** tests ;)
> 
Cool!:)

crazyscientist commented:
sorry, my bad, did not notice the blank line at the end.

I can't squash the commits, but according to https://help.github.com/articles/about-pull-request-merges/#squash-and-merge-your-pull-request-commits you should be able to squash the commits while accepting the pull request.

HaraldMuellerNey commented:
It seems we reach agrrement on the MR, or I am wrong? What is missing to get it merged?

crazyscientist commented:
I think, @marcus-h, did not believe me that the commits could be squashed while accepting the pull request. I have rebased and force pushed. Now this should be good to be merged/pulled.

marcus-h commented:
It seems github ignores my mails again. Here's the forwarded message:
```
----- Forwarded message from Marcus Hüwe <suse-tux@gmx.de> -----
  
Date: Wed, 14 Nov 2018 21:29:49 +0100
From: Marcus Hüwe <suse-tux@gmx.de>
To: reply+00104a4c17133a55047e64cb480785e527fec0bc74364ccd92cf000000011803b42d92a169ce14c28fe6@reply.github.com
Subject: Re: [openSUSE/osc2] Added support to search for packages (#32)
User-Agent: NeoMutt/20180716

On 2018-11-14 02:00:13 -0800, Andreas Hasenkopf wrote:
> I think, @marcus-h, did not believe me that the commits could be squashed while accepting the pull request. I have rebased and force pushed. Now this should be good to be merged/pulled.
> 
Thanks! Merged.

Thanks a lot for the PR!

----- End forwarded message -----
```
